File: programmers/etc/level2/연속된_부분_수열의_합.py
def solution(sequence, k):
    answer = []
    n = len(sequence)
    tmp = sequence[0]
    left, right = 0, 0
    while left <= right:
        if tmp < k:
            right += 1
            if right == n:
                break
            tmp += sequence[right]
        elif tmp > k:
            tmp -= sequence[left]
            left += 1
        else:
            answer.append([right-left+1, left, right])
            tmp -= sequence[left]
            left += 1; right += 1
            if right == n:
                break
            tmp += sequence[right]
    answer.sort()
    return answer[0][1:]


sequence = [1, 2, 3, 4, 5]
k = 7
print(solution(sequence, k)) # [2,3]

sequence = [1, 1, 1, 2, 3, 4, 5]
k = 5
print(solution(sequence, k)) # [6,6]

sequence = [2, 2, 2, 2, 2]
k = 6
print(solution(sequence, k)) # [0,2]

File: Java/programmers/etc/level2/연속된부분수열의합.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 연속된부분수열의합 {
    public int[] solution(int[] sequence, int k) {
        List<int[]> answer = new ArrayList<>();
        int tot = 0, left = 0, right = 0;
        int n = sequence.length;

        while (left <= right && right < n) {
            tot += sequence[right];
            if (tot >= k) {
                while (tot > k) {
                    tot -= sequence[left];
                    left++;
                }
                if (tot == k) answer.add(new int[]{left, right});
            }
            right++;
        }

        answer.sort((o1, o2) -> {
            int d1 = o1[1]-o1[0];
            int d2 = o2[1]-o2[0];
            return d1 == d2 ? o1[0]-o2[0] : (d1-d2);
        });

        return answer.get(0);
    }
}
